Bottom: 4951bf3c312c4cecdaef76cf02871ab8890ea72e
Top:    e9d3aa1166e2dcd28631505ca2e7c3e09b57eac9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 15:17:52 +0100

Remove SWAP_IDENT, pass it as an argument where needed

Magic state is bad ;)


---

diff --git a/src/draw.c b/src/draw.c
index 0a99bbc..706f8bd 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -339,8 +339,6 @@ rat_callback (const BoxType * b, void *cl)
 static void
 PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
 {
-  int save_swap = SWAP_IDENT;
-
   gui->set_draw_faded (Output.fgGC, 1);
   SWAP_IDENT = swap_ident;
   DrawLayerGroup (side_group, drawn_area);
@@ -351,7 +349,6 @@ PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
   DrawSilk (swap_ident,
 	    swap_ident ? solder_silk_layer : component_silk_layer,
 	    drawn_area);
-  SWAP_IDENT = save_swap;
 }
 
 /* ---------------------------------------------------------------------------
@@ -413,8 +410,6 @@ DrawEverything (BoxTypePtr drawn_area)
 	{
 	  if (DrawLayerGroup (group, drawn_area) && !gui->gui)
 	    {
-	      int save_swap = SWAP_IDENT;
-
 	      if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
 		continue;
 	      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback,
@@ -428,7 +423,6 @@ DrawEverything (BoxTypePtr drawn_area)
 		  r_search (PCB->Data->pad_tree, drawn_area, NULL,
 			    pad_callback, NULL);
 		}
-	      SWAP_IDENT = save_swap;
 
 	      if (!gui->gui)
 		{
@@ -1966,8 +1960,6 @@ hid_expose_callback (HID * hid, BoxType * region, void *item)
 
   Gathering = false;
 
-  /*printf("\033[32mhid_expose_callback, s=%p %d\033[0m\n", &(SWAP_IDENT), SWAP_IDENT); */
-
   hid->set_color (Output.pmGC, "erase");
   hid->set_color (Output.bgGC, "drill");
 
diff --git a/src/macro.h b/src/macro.h
index 19f80b8..95236af 100644
--- a/src/macro.h
+++ b/src/macro.h
@@ -34,11 +34,7 @@
 
 /* ---------------------------------------------------------------------------
  * macros to transform coord systems
- * draw.c uses a different definition of TO_SCREEN
  */
-#ifndef	SWAP_IDENT
-#define	SWAP_IDENT			Settings.ShowSolderSide
-#endif
 
 #define	SWAP_SIGN_X(x)		(x)
 #define	SWAP_SIGN_Y(y)		(-(y))
